integ/link_test.go

```diff
@@ -79,11 +90,17 @@ func TestCreateLink(t *testing.T) {
 				Expect(t).Status(tcase.ExpectStatus)
 
 			if tcase.ExpectStatus < http.StatusBadRequest {
-				tt.Assert(jsonpath.Equal("$.link.url", tcase.GivenBody["url"]))
-				tt.Assert(jsonpath.Equal("$.link.title", tcase.GivenBody["title"]))
-				tt.Assert(jsonpath.Equal("$.link.description", tcase.GivenBody["description"]))
-				tt.Assert(jsonpath.Equal("$.link.favicon", tcase.GivenBody["favicon"]))
-				tt.Assert(jsonpath.Equal("$.link.site", tcase.GivenBody["site"]))
+				if !tcase.SkipValueCheck {
+					tt.Assert(jsonpath.Equal("$.link.url", tcase.GivenBody["url"]))
+					tt.Assert(jsonpath.Equal("$.link.title", tcase.GivenBody["title"]))
+					tt.Assert(jsonpath.Equal("$.link.description", tcase.GivenBody["description"]))
+					tt.Assert(jsonpath.Equal("$.link.favicon", tcase.GivenBody["favicon"]))
+					tt.Assert(jsonpath.Equal("$.link.site", tcase.GivenBody["site"]))
+				} else {
+					// Just check that link was created successfully
+					tt.Assert(jsonpath.Present("$.link.id"))
+					tt.Assert(jsonpath.Equal("$.link.url", tcase.GivenBody["url"]))
+				}
```

Comment:
The addition of `SkipValueCheck` is not necessary since, in the test environment, the `Analyzer` dependency is mocked and always returns certain values. Here's the relevant code from `analyze.go`:

```
func (c *TestClient) Do(ctx context.Context, req *Request) (*Response, error) {
	return &Response{
		URL:         req.URL,
		Image:       "https://via.placeholder.com/150",
		Favicon:     "https://via.placeholder.com/16",
		Title:       "Testing",
		Site:        "testing.com",
		Description: "It's only a test.",
		Original:    req.URL,
		Corpus:      "It's only a test.",
	}, nil
}
```

So you can get rid of `SkipValueCheck` and use the values here.

Reply:
Instead of relying on `tcase.SkipValueCheck`, which should be removed, assert that `jsonpath.Equal("$.link.favicon", "https://via.placeholder.com/16")`. This should work because the mocked Analyzer always returns `"https://via.placeholder.com/16"` as the favicon.